htlcswitch/failure.go

```diff
@@ -262,10 +262,14 @@ func (s *SphinxErrorDecrypter) DecryptError(reason lnwire.OpaqueReason) (*Forwar
 		return nil, err
 	}
 
+	// Decode the failure. If an error occurs, we leave the failure message
```

Comment:
why?

Reply:
Because it is not an internal error. We do have an error source and we can only process this outcome in a proper way in mission control if we can distinguish this case from internal errors.